unit-tests/src/main/scala/scala/scalanative/native/InstanceOfSuite.scala

```diff
@@ -0,0 +1,79 @@
+package scala.scalanative.native
+
+object InstanceOfSuite extends tests.Suite {
+
+  // asInstanceOf
+
+//  test("expects anyRef.asInstanceOf[String] fail") {
```

Comment:
That's good, `asInstanceOf` has a bunch of other corner cases we don't fully handle right now, apart from null. I'm going to look at this today. This definitely should go into a separate PR.